# Statistical Tests Assignment 

In [1]:
import pandas as pd
import numpy as np
import scipy.stats as stats
import statsmodels as statmodel

><em>1. A physician is evaluating a new diet for her patients with a family history of heart disease. To test the effectiveness of this diet, 16 patients are placed on the diet for 6 months. Their weights and triglyceride levels are measured before and after the study, and the physician wants to know if either set of measurements has changed. (Data set: dietstudy.csv)</em>

In [5]:
dietdf=pd.read_csv('dietstudy.csv')
dietdf

patid  age  gender  tg0  tg1  tg2  tg3  tg4  wgt0  wgt1  wgt2  wgt3  wgt4
0       1   45    Male  180  148  106  113  100   198   196   193   188   192
1       2   56    Male  139   94  119   75   92   237   233   232   228   225
2       3   50    Male  152  185   86  149  118   233   231   229   228   226
3       4   46  Female  112  145  136  149   82   179   181   177   174   172
4       5   64    Male  156  104  157   79   97   219   217   215   213   214
5       6   49  Female  167  138   88  107  171   169   166   165   162   161
6       7   63    Male  138  132  146  143  132   222   219   215   215   210
7       8   63  Female  160  128  150  118  123   167   167   166   162   161
8       9   52    Male  107  120  129  195  174   199   200   196   196   193
9      10   45    Male  156  103  126  135   92   233   229   229   229   226
10     11   61  Female   94  144  114  114  121   179   181   176   173   173
11     12   49  Female  107   93  156  148  150   158   153   155   155   154
12     13   61  Female  145  107  129   86  159   157   151   150   145   143
13     14   59    Male  186  142  128  122  101   216   213   210   210   206
14     15   52    Male  112  107  103   89  148   257   255   254   252   249
15     16   60  Female  104  103  117   79  130   151   146   144   144   140

<pre>No CLT (since sample size < 30)

paired t-test will be applied on both
significance value=99%

case1:
H0: mean(tg before)=mean(tg after)
HA: mean(tg before)!=mean(tg after)

case1:
H0: mean(wt before)=mean(wt after)
HA: mean(wt before)!=mean(wt after)

df=15
critical interval= 99%
2 tailed test
a=0.01
a/2=0.005
ta/2=t(0.005,15)=2.95

</pre>

In [89]:
stats.ttest_rel(dietdf['tg0'],dietdf['tg4'])

Ttest_relResult(statistic=1.2000008533342437, pvalue=0.24874946576903698)

$$pvalue/2 > 0.01$$
$$t=1.2 < ta/2$$
>><pre>
we could not reject the null hypothesis with 99% significance level
thus, mean(tg before)=mean(tg after)
</pre>

In [91]:
stats.ttest_rel(dietdf['wgt0'],dietdf['wgt4'])

Ttest_relResult(statistic=11.174521688532522, pvalue=1.137689414996614e-08)


 $$pvalue/2 < 0.01$$
$$t=11 > ta/2$$
>><pre>we reject the null hypothesis,H0  with 99% significance level
an accept HA
thus, mean(wt before)!=mean(wt after)
</pre>

><em>2. An analyst at a department store wants to evaluate a recent credit card promotion. To this end, 500 cardholders were randomly selected. Half received an ad promoting a reduced interest rate on purchases made over the next three months, and half received a standard seasonal ad. Is the promotion effective to increase sales? (Data set: creditpromo.csv)

<pre>CLT (samples > 30) but no std given
H0: mean(standard) = mean(prom)
HA: mean(standard) != mean(prom)
test : 2 sample t-test(Independent)
significance value : 99%

</pre>

In [92]:
credit=pd.read_csv("creditpromo.csv")
credit.head()

id         insert      dollars
0   148       Standard  2232.771979
1   572  New Promotion  1403.807542
2   973       Standard  2327.092181
3  1096       Standard  1280.030541
4  1541  New Promotion  1513.563200

In [93]:
standard=credit['dollars'][credit['insert']=='Standard']
newpro=credit['dollars'][credit['insert']=="New Promotion"]

In [94]:
standard.count(),newpro.count()

(250, 250)

<pre>
critical interval= 99%
2 tailed test

a=0.01
a/2=0.005
ta/2=t(0.005,249)=2.6

reject H0 if,
t>ta/2
</pre>

In [95]:
stats.ttest_ind(standard,newpro)

Ttest_indResult(statistic=-2.2604227264649963, pvalue=0.024225996894147814)


 $$pvalue/2 > 0.01$$
$$|t|=2.26 < ta/2$$
>><pre>
we could not reject null hypothesis,H0  with 99% significance level
thus,we can't say that promotion was effective to increase sales
</pre>

><em>3. An experiment is conducted to study the hybrid seed production of bottle gourd under open field conditions. The main aim of the investigation is to compare natural pollination and hand pollination. The data are collected on 10 randomly selected plants from each of natural pollination and hand pollination. The data are collected on fruit weight (kg), seed yield/plant (g) and seedling length (cm). (Data set: pollination.csv)
<br>a. Is the overall population of Seed yield/plant (g) equals to 200?
<br>b. Test whether the natural pollination and hand pollination under open field conditions are equally effective or are significantly different.

In [96]:
poll=pd.read_csv('pollination.csv')
poll

Group  Fruit_Wt  Seed_Yield_Plant  Seedling_length
0   Natural      1.85            147.70            16.86
1   Natural      1.86            136.86            16.77
2   Natural      1.83            149.97            16.35
3   Natural      1.89            172.33            18.26
4   Natural      1.80            144.46            17.90
5   Natural      1.88            138.30            16.95
6   Natural      1.89            150.58            18.15
7   Natural      1.79            140.99            18.86
8   Natural      1.85            140.57            18.39
9   Natural      1.84            138.33            18.58
10     Hand      2.58            224.26            18.18
11     Hand      2.74            197.50            18.07
12     Hand      2.58            230.34            19.07
13     Hand      2.62            217.05            19.00
14     Hand      2.68            233.84            18.00
15     Hand      2.56            216.52            18.49
16     Hand      2.34            211.93            17.45
17     Hand      2.67            210.37            18.97
18     Hand      2.45            199.87            19.31
19     Hand      2.44            214.30            19.36

<pre>No CLT (samples < 30) 
H0: mean(Seed yield/plant) = 200
HA: mean(Seed yield/plant) != 200 
test : 1 sample t-test
significance value : 99%
</pre>


In [97]:
stats.ttest_1samp(poll.Seed_Yield_Plant,200)

Ttest_1sampResult(statistic=-2.3009121248548645, pvalue=0.032891040921283025)

 $$pvalue/2 > 0.01$$
$$|t| = 2.3 < ta/2$$
>><pre>we can not reject the null hypothesis,H0  with 99% significance level
thus we <b>cannot</b> say that overall population of seed_yield/plant != 200
</pre>

In [98]:
poll.Seed_Yield_Plant.mean()

180.8035

<pre>

2 tail test 

Case 1 :
H0: mean(hand poll Fruit_Wt) = mean(natural poll Fruit_Wt)
HA: mean(hand poll Fruit_Wt) != mean(natural poll Fruit_Wt)
test : 2 sample t-test(Independent)
significance value : 99%

Case 2 :
H0: mean(hand poll Seed_Yield_Plant) = mean(natural poll Seed_Yield_Plant)
HA: mean(hand poll Fruit_Wt) != mean(natural poll Fruit_Wt)
test : 2 sample t-test(Independent)
significance value : 99%

Case 3 :
H0: mean(hand poll Seedling_len) = mean(natural poll Seedling_len)
HA: mean(hand poll Seedling_len) != mean(natural poll Seedling_len)
test : 2 sample t-test(Independent)
significance value : 99%


df=9
a=0.01
a/2=0.005
ta/2=3.2

reject null hypothesis if,
|t|>ta/2

</pre>


In [99]:
f1 = poll['Fruit_Wt'][poll.Group == 'Natural']
f2 = poll['Fruit_Wt'][poll.Group == 'Hand']

In [100]:
stats.ttest_ind(f1, f2,equal_var=False)

Ttest_indResult(statistic=-17.669989614440286, pvalue=4.306871213074868e-09)

$$pvalue<0.01$$

$$ |t|=17.6>ta/2 $$

>><pre>
we reject H0 with 99% significance.
Thus,there is significant difference for natural and hand pollination
<pre>

In [101]:
s1 = poll['Seed_Yield_Plant'][poll.Group == 'Natural']
s2 = poll['Seed_Yield_Plant'][poll.Group == 'Hand']

In [102]:
stats.ttest_ind(s1,s2,equal_var=False)

Ttest_indResult(statistic=-13.958260515902547, pvalue=5.136161282685624e-11)

$$pvalue<0.01$$

$$ |t|=13.9>ta/2 $$

>><pre>
we reject H0 with 99% significance.
Thus,there is significant difference for natural and hand pollination
<pre>

In [103]:
sl1 = poll['Seedling_length'][poll.Group == 'Natural']
sl2 = poll['Seedling_length'][poll.Group == 'Hand']

In [104]:
stats.ttest_ind(sl1,sl2,equal_var=False)

Ttest_indResult(statistic=-2.542229999657055, pvalue=0.021430608378161634)

$$pvalue>0.01$$

$$ |t|=2.5|>ta/2 $$

>><pre>
we <b>cannot</b> reject H0 with 99% significance.
Thus,there is <b>not</b> significant difference for natural and hand pollination
<pre>

><em>4. An electronics firm is developing a new DVD player in response to customer requests. Using a prototype, the marketing team has collected focus data for different age groups viz. Under 25; 25-34; 35-44; 45-54; 55-64; 65 and above. Do you think that consumers of various ages rated the design differently? (Data set: dvdplayer.csv).

In [24]:
dvd=pd.read_csv('dvdplayer.csv')
dvd.head()

agegroup   dvdscore
0  65 and over  38.454803
1        55-64  17.669677
2  65 and over  31.704307
3  65 and over  25.924460
4     Under 25  30.450007

In [121]:
dvd.agegroup.value_counts()

65 and over    17
Under 25       13
35-44          12
45-54          10
25-34          10
55-64           6
Name: agegroup, dtype: int64

<pre>
H0: There is similar difference Between and within the groups
HA: There is more difference between groups than within groups

One-Way Anova Test 
    
</pre>

In [125]:
d1=dvd['dvdscore'][dvd['agegroup']=='65 and over']
d2=dvd['dvdscore'][dvd['agegroup']=='Under 25']
d3=dvd['dvdscore'][dvd['agegroup']=='35-44']
d4=dvd['dvdscore'][dvd['agegroup']=='45-54']
d5=dvd['dvdscore'][dvd['agegroup']=='25-34']
d6=dvd['dvdscore'][dvd['agegroup']=='55-64']

<pre>
no of var,c=6
no of obs,n=68
df1=c-1=5
df2=n-c=62
a=0.01
Fa=9.2

reject H0,if
F > Fa
</pre>

In [126]:
stats.f_oneway(d1,d2,d3,d4,d5,d6)

F_onewayResult(statistic=6.992526962676517, pvalue=3.087324905679639e-05)

$$pvalue>0.01$$
$$F<Fa$$
>><pre>
We cannot reject the H0,
thus ,we can say that comsumers of different ages have <b>not rated differently</b>
</pre>

><em>5. A survey was conducted among 2800 customers on several demographic characteristics. Working status, sex, age, age-group, race, happiness, no. of child, marital status, educational qualifications, income group etc. had been captured for that purpose. (Data set: sample_survey.csv).
<br>a. Is there any relationship in between labour force status with marital status?
<br>b. Do you think educational qualification is somehow controlling the marital status?
<br>c. Is happiness is driven by earnings or marital status?

In [23]:
surv=pd.read_csv('sample_survey.csv')
surv.head()

id            wrkstat        marital  childs   age  educ  paeduc  maeduc  \
0   1  Working full time       Divorced     2.0  60.0  12.0    12.0    12.0   
1   2  Working part-time  Never married     0.0  27.0  17.0    20.0     NaN   
2   3  Working full time        Married     2.0  36.0  12.0    12.0    12.0   
3   4  Working full time  Never married     0.0  21.0  13.0     NaN    12.0   
4   5  Working full time  Never married     0.0  35.0  16.0     NaN    12.0   

   speduc          degree    ...           agecat childcat news1 news2  news3  \
0     NaN     High school    ...         55 to 64      1-2    No    No     No   
1     NaN  Junior college    ...         25 to 34     None    No    No    Yes   
2    16.0     High school    ...         35 to 44      1-2    No    No     No   
3     NaN     High school    ...     Less than 25     None    No    No     No   
4     NaN        Bachelor    ...         35 to 44     None    No    No     No   

  news4 news5      car1      car2      car3  
0    No    No  American  Japanese  Japanese  
1    No    No  American    German  Japanese  
2   Yes   Yes  American  American       NaN  
3   Yes   Yes  American     Other       NaN  
4    No    No  American  American    Korean  

[5 rows x 46 columns]

<pre>
case 1:
H0: Two categorical variables are independent.
HA: Two categorical variables are not indeoendent.

chi-square test will be applied.
for wrkstat,marital status

In [14]:
work=pd.crosstab(surv.marital,surv.wrkstat)
work

wrkstat        Keeping house  Other  Retired  School  Temporarily not working  \
marital                                                                         
Divorced                  25     12       53       7                        9   
Married                  200     16      168       9                       23   
Never married             35     14       17      60                       11   
Separated                 13      4        6       2                        1   
Widowed                   55      8      150       1                        2   

wrkstat        Unemployed, laid off  Working full time  Working part-time  
marital                                                                    
Divorced                         10                295                 35  
Married                          13                778                138  
Never married                    32                392                102  
Separated                         0                 58                  9  
Widowed                           3                 44                 20

In [15]:
stats.chi2_contingency(observed= work)

(729.2421426572284,
 1.4875268409067568e-135,
 28,
 array([[ 51.69187279,   8.51024735,  62.09328622,  12.45017668,
           7.24946996,   9.14063604, 246.95477032,  47.90954064],
        [155.8869258 ,  25.66431095, 187.25441696,  37.5459364 ,
          21.86219081,  27.56537102, 744.74028269, 144.48056537],
        [ 76.84240283,  12.65088339,  92.30459364,  18.50777385,
          10.77667845,  13.58798587, 367.10989399,  71.21978799],
        [ 10.77879859,   1.7745583 ,  12.94770318,   2.59611307,
           1.51166078,   1.90600707,  51.495053  ,   9.99010601],
        [ 32.8       ,   5.4       ,  39.4       ,   7.9       ,
           4.6       ,   5.8       , 156.7       ,  30.4       ]]))

$$pvalue < 0.01$$
>><pre>I rejected the null and there is relation between labour force status with marital status</pre>

<pre>
case 2:
H0: Two categorical variables are independent.
HA: Two categorical variables are not indeoendent.

chi-square test will be applied.
for marital status,degree

In [35]:
edu=pd.crosstab(surv.degree,surv.marital)
edu

marital         Divorced  Married  Never married  Separated  Widowed
degree                                                              
Bachelor              58      251            129         12       28
Graduate              29      123             41          3        9
High school          241      686            367         58      148
Junior college        45      108             46          3        6
LT High school        70      174             77         17       92

In [36]:
stats.chi2_contingency(observed= edu)

(122.68449020508541,
 1.6707923432360119e-18,
 16,
 array([[ 75.06345268, 227.39312301, 111.83268345,  15.75824176,
          47.95249911],
        [ 32.19248493,  97.52215526,  47.9617157 ,   6.75824176,
          20.56540234],
        [235.55476781, 713.57674583, 350.9393832 ,  49.45054945,
         150.4785537 ],
        [ 32.66359447,  98.94930876,  48.66359447,   6.85714286,
          20.86635945],
        [ 67.52570011, 204.55866714, 100.60262318,  14.17582418,
          43.1371854 ]]))

$$pvalue < 0.01$$
>><pre>I rejected the null and there is relation between educational qualifiaction status with marital status</pre>

<pre>
case 3:
H0: Two categorical variables are independent.
HA: Two categorical variables are not indeoendent.

chi-square test will be applied.
a)for happy,earning
b)for happy,marital status

In [22]:
earn=pd.crosstab(surv.happy,surv.income)
earn

income         $1000 TO 2999  $10000 - 14999  $15000 - 19999  $20000 - 24999  \
happy                                                                          
Not too happy              7              39              33              40   
Pretty happy              20             107             119             155   
Very happy                 5              44              26              50   

income         $25000 or more  $3000 TO 3999  $4000 TO 4999  $5000 TO 5999  \
happy                                                                        
Not too happy             113              9              9              6   
Pretty happy              888             11             13             18   
Very happy                571              4             10             11   

income         $6000 TO 6999  $7000 TO 7999  $8000 TO 9999  LT $1000  
happy                                                                 
Not too happy             14             12              9        11  
Pretty happy              13             21             30        13  
Very happy                 6             14             19        11

In [19]:
stats.chi2_contingency(observed= earn)

(178.9505306121643,
 1.4107677273473057e-26,
 22,
 array([[  3.89520355,  23.12777106,  21.66706973,  29.82265216,
         191.35187424,   2.92140266,   3.89520355,   4.26037888,
           4.01692866,   5.72108021,   7.06005643,   4.26037888],
        [ 18.16041919, 107.82748892, 101.01733172, 139.04070939,
         892.1305925 ,  13.62031439,  18.16041919,  19.86295848,
          18.72793229,  26.67311568,  32.91575977,  19.86295848],
        [  9.94437727,  59.04474002,  55.31559855,  76.13663845,
         488.51753325,   7.45828295,   9.94437727,  10.87666264,
          10.25513906,  14.60580411,  18.0241838 ,  10.87666264]]))

$$p<0.01$$
H0 cannot be accepted.
thus the income and happiness are in relation

In [20]:
marital=pd.crosstab(surv.happy,surv.marital)
marital

marital        Divorced  Married  Never married  Separated  Widowed
happy                                                              
Not too happy        72       71            108         30       59
Pretty happy        278      684            426         49      137
Very happy           93      582            120         13       83

In [21]:
stats.chi2_contingency(marital)

(260.6894389418282,
 9.3147261197964e-52,
 8,
 array([[ 53.6969697 , 162.06060606,  79.27272727,  11.15151515,
          33.81818182],
        [248.58538324, 750.24527629, 366.98609626,  51.62495544,
         156.55828877],
        [140.71764706, 424.69411765, 207.74117647,  29.22352941,
          88.62352941]]))

$$p<0.01$$
thus H0 cannit be accepted.
thus,happiness and marital status are related

>><pre>thus,Hapiness is driven by both:marital status and income